.github/workflows/main.yaml

```diff
@@ -25,14 +25,23 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create branch from commit by event name
+        run: |
+          if [[ '${{ github.event_name }}' == 'push' ]]; then
+            git branch __branch_before ${{ github.event.before }}
+          elif [[ "${{ github.event_name }}" == 'pull_request' ]]; then
+            git branch __branch_before ${{ github.event.pull_request.base.sha }}
+          fi
       - name: Set up JDK ${{ matrix.java_version }}
         uses: actions/setup-java@v4
         with:
           java-version: ${{ matrix.java_version }}
           distribution: 'temurin'
           cache: 'maven'
       - name: Build with JDK ${{ matrix.java_version }}
-        run: mvn --batch-mode install -DskipITs ${{ matrix.included_modules }}
+        run: mvn --batch-mode install ${{ matrix.included_modules }} -Dgib.disable=false -Dgib.buildDownstream=false -Dgib.buildUpstream=true -Dgib.referenceBranch=__branch_before
```

Comment:
I'm surprised by `buildDownstream=false`. This means that you won't notice if B (which depends on A) breaks because of some incompatible change in A.
This is especially surprising because this `mvn` call does not seem to be only about ITs.

Reply:
hi @famod ,thanks for your review

with some research before
i found that
- there 3 ```workflow``` realtive to this issues
- [main.yaml](https://github.com/langchain4j/langchain4j/blob/main/.github/workflows/main.yaml) for ensure compile sucess and test passed when ```commits``` or ```pull request``` push to ```main``` 
- [snapshot_release.yaml](https://github.com/langchain4j/langchain4j/blob/main/.github/workflows/snapshot_release.yaml) for build and deploy all modules to remote repository except ```langchain4j-core``` and ```langchain4j-parent``` when commits push to branch ```main```
- [snapshot_release_core_and_parent.yaml](https://github.com/langchain4j/langchain4j/blob/main/.github/workflows/snapshot_release_core_and_parent.yaml) for build and deploy module ```langchain4j-core``` and ```langchain4j-parent``` to remote repository
- due to the [comment](https://github.com/langchain4j/langchain4j/pull/1185#issuecomment-2136771673), @langchain4j may want to the submodule build、test if we could achive that
- if we skip module dependent by other(e.g. langchain4j-core),with the help of ```snapshot_release.yaml``` and ```snapshot_release_core_and_parent.yaml```,we could fetch them from remote instead of build them locally

so i set ```buildDownstream=false``` explicitly,to avoid submodule build
@famod should i remove them?i am not really sure my usage is correct here

Reply:
@famod 
after reread the doc
i found i make a stupid mistake
i reversved the meaning of UpstreamModule and DownstreamModule
the default property value of ```gib.buildDownstream``` and ```gib.buildUpstream``` is the behaviour i want
i will make a new commit again

could you help me to check it again?thks